zml/context.zig

```diff
@@ -121,13 +127,13 @@ pub const Context = struct {
     pub fn autoPlatform(self: *Context) Platform {
         // the last platform is the one that with the high enum number, so considered
         // to be the "best" one
-        var platform_: Platform = undefined;
+        var platform_: ?Platform = null;
         var iterator = self.platforms.iterator();
         while (iterator.next()) |entry| {
             if (entry.value.*) |p| {
                 platform_ = p;
             }
         }
-        return platform_;
+        return platform_ orelse @panic("No platform found !");
```

Comment:
I nearly panicked over this. But: no platform means: not even a CPU platform -> ZML cannot work -> should be OK-ish to panic. I would prefer an error though and decide myself if I want to panic or send off an alarm to "monitoring" and terminate, etc.

One fictional scenario I can come up with: some code that optionally calls a ZML model, e.g. some web app that server-side gives some LLM hints or whatever. It might want to start and check if it can run inference - and ignore ZML if it can't. In that scenario, a panicking autoplatform() is bad. However, there's a remedy: you can always check the platforms.iterator() yourself and see if you get at least one iteration out of it.

The implication of introducing an error here, is: we'd need to change the examples and tell people. Given how early we are, I wouldn't mind breaking the API for the (in my point of view) better.

Since no platform "should never happen", I'm happy to discuss introducing the error at a later time, in order to not block this PR.

Reply:
so `zml.contex.init()` already return you an error where there is no platform.
here the panic happen if someone created a `zml.Context` manually, or if someone removed all platforms manually.
I think it's reasonable to panic here.

Reply:
OK, thanks for that detail. I agree.